extractor/src/main/java/org/schabi/newpipe/extractor/utils/ExtractorLogger.java

```diff
@@ -0,0 +1,48 @@
+package org.schabi.newpipe.extractor.utils;
+
+public final class ExtractorLogger {
+
+    private ExtractorLogger() { }
+
+    private static Logger logger = new ConsoleLogger();
+
+    public static void setLogger(final Logger customLogger) {
+        logger = customLogger;
+    }
+
+    public static void d(final String tag, final String msg) {
+        logger.debug(tag, msg);
+    }
+
+    public static void w(final String tag, final String msg) {
+        logger.warn(tag, msg);
+    }
+
+    public static void e(final String tag, final String msg) {
+        logger.error(tag, msg);
+    }
+
+    public static void e(final String tag, final String msg, final Throwable t) {
+        logger.error(tag, msg, t);
+    }
+
+    // default logger that prints to stdout
+    private static final class ConsoleLogger implements Logger {

```

Comment:
This should be only set by users of the library and should default to a logger that logs nothing. In each extractor test then you would set this `ConsoleLogger` as the logger (i.e. you just add this in the common tests initialization methods).